### ISSUE TITLE
Dlib installation failor and coral dev board, creatig swap file on microsd consuming from root memory. 

### ISSUE THREAD

**Issue by cappittall** [goruck/edge-tpu-servers#12]
Hello @goruck ,
Thanks in advace for your nice guiding repo. 
I would like to ask that when I follow the instruction for swapfile. I mounted sd card on boot with writing to /etc/fstab
and mounted good. When I follow : 
$ cd /media/mendel

```
# Create a swapfile else you'll run out of memory compiling.
$ sudo mkdir swapfile
# Now let's increase the size of swap file.
$ sudo dd if=/dev/zero of=/swapfile bs=1M count=2048 oflag=append conv=notrunc
# Setup the file as a "swap file".
$ sudo mkswap /swapfile
# Enable swapping.
$ sudo swapon /swapfile
```

Then after swapfile seams at root and consuming root memory instead sd card. What am I doing wrong. 
Here the result disk usage after swap file (before swap file root was %40)

mendel@green-zebra:/media/mendel$ df
Filesystem     1K-blocks    Used Available Use% Mounted on
/dev/root        5230192 3955464   1001120  80% /
devtmpfs          340316       0    340316   0% /dev
tmpfs             504796       0    504796   0% /dev/shm
tmpfs             504796   19448    485348   4% /run
tmpfs               5120       4      5116   1% /run/lock
tmpfs             504796       0    504796   0% /sys/fs/cgroup
tmpfs             504796     484    504312   1% /var/log
/dev/mmcblk0p3   1998672  147668   1729764   8% /home
/dev/mmcblk0p1    126931   30508     89870  26% /boot
tmpfs             100956   13052     87904  13% /run/user/1000
/dev/sda        15133808 1065480  14068328   8% /media/microsd
/dev/mmcblk1    30546356   45084  28926540   1% /media/mendel

Also /swapfile folder exist in root too

is this normal or something I did wrong?

Also I am not able to build dlib since 3 days :)) gives error like : 

```
mendel@green-zebra:/media/mendel/dlib$ python3 setup.py install --set DLIB_NO_GUI_SUPPORT=YES \
> --set DLIB_USE_CUDA=NO --compiler-flags "-O3"
running install
running bdist_egg
running egg_info
creating tools/python/dlib.egg-info
error: could not create 'tools/python/dlib.egg-info': Permission denied
mendel@green-zebra:/media/mendel/dlib$ sudo python3 setup.py install --set DLIB_NO_GUI_SUPPORT=YES --set DLIB_USE_CUDA=NO --compiler-flags "-O3"
running install
running bdist_egg
running egg_info
creating tools/python/dlib.egg-info
writing tools/python/dlib.egg-info/PKG-INFO
writing dependency_links to tools/python/dlib.egg-info/dependency_links.txt
writing top-level names to tools/python/dlib.egg-info/top_level.txt
writing manifest file 'tools/python/dlib.egg-info/SOURCES.txt'
package init file 'tools/python/dlib/__init__.py' not found (or not a regular file)
reading manifest file 'tools/python/dlib.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
no previously-included directories found matching 'tools/python/build*'
writing manifest file 'tools/python/dlib.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-aarch64/egg
running install_lib
running build_py
running build_ext
Building extension for Python 3.7.3 (default, Jan 22 2021, 20:04:44) 
Invoking CMake setup: 'cmake /media/mendel/dlib/tools/python -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/media/mendel/dlib/build/lib.linux-aarch64-3.7 -DPYTHON_EXECUTABLE=/usr/bin/python3 -DDLIB_NO_GUI_SUPPORT=YES -DDLIB_USE_CUDA=NO -DCMAKE_CXX_FLAGS=-O3 -DCMAKE_BUILD_TYPE=Release'
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: /usr/bin/python3 (found version "3.7.3") 
-- Found PythonLibs: python3.7m
-- Performing Test HAS_CPP14_FLAG
-- Performing Test HAS_CPP14_FLAG - Success
-- pybind11 v2.2.4
-- Using CMake version: 3.13.4
-- Compiling dlib version: 19.22.99
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Searching for BLAS and LAPACK
-- Searching for BLAS and LAPACK
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Looking for cblas_ddot
-- Looking for cblas_ddot - not found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of void*
-- Check size of void* - done
 *****************************************************************************
 *** No BLAS library found so using dlib's built in BLAS.  However, if you ***
 *** install an optimized BLAS such as OpenBLAS or the Intel MKL your code ***
 *** will run faster.  On Ubuntu you can install OpenBLAS by executing:    ***
 ***    sudo apt-get install libopenblas-dev liblapack-dev                 ***
 *** Or you can easily install OpenBLAS from source by downloading the     ***
 *** source tar file from http://www.openblas.net, extracting it, and      ***
 *** running:                                                              ***
 ***    make; sudo make install                                            ***
 *****************************************************************************
-- C++11 activated.
-- Configuring done
-- Generating done
-- Build files have been written to: /media/mendel/dlib/build/temp.linux-aarch64-3.7
Invoking CMake build: 'cmake --build . --config Release -- -j1'
Scanning dependencies of target dlib
[  1%] Building CXX object dlib_build/CMakeFiles/dlib.dir/base64/base64_kernel_1.cpp.o
[  2%] Building CXX object dlib_build/CMakeFiles/dlib.dir/bigint/bigint_kernel_1.cpp.o
[  2%] Building CXX object dlib_build/CMakeFiles/dlib.dir/bigint/bigint_kernel_2.cpp.o
[  3%] Building CXX object dlib_build/CMakeFiles/dlib.dir/bit_stream/bit_stream_kernel_1.cpp.o
[  4%] Building CXX object dlib_build/CMakeFiles/dlib.dir/entropy_decoder/entropy_decoder_kernel_1.cpp.o
[  4%] Building CXX object dlib_build/CMakeFiles/dlib.dir/entropy_decoder/entropy_decoder_kernel_2.cpp.o
[  5%] Building CXX object dlib_build/CMakeFiles/dlib.dir/entropy_encoder/entropy_encoder_kernel_1.cpp.o
[  5%] Building CXX object dlib_build/CMakeFiles/dlib.dir/entropy_encoder/entropy_encoder_kernel_2.cpp.o
[  6%] Building CXX object dlib_build/CMakeFiles/dlib.dir/md5/md5_kernel_1.cpp.o
[  7%] Building CXX object dlib_build/CMakeFiles/dlib.dir/tokenizer/tokenizer_kernel_1.cpp.o
[  7%] Building CXX object dlib_build/CMakeFiles/dlib.dir/unicode/unicode.cpp.o
[  8%] Building CXX object dlib_build/CMakeFiles/dlib.dir/test_for_odr_violations.cpp.o
[  9%] Building CXX object dlib_build/CMakeFiles/dlib.dir/sockets/sockets_kernel_1.cpp.o
[  9%] Building CXX object dlib_build/CMakeFiles/dlib.dir/bsp/bsp.cpp.o
[ 10%] Building CXX object dlib_build/CMakeFiles/dlib.dir/dir_nav/dir_nav_kernel_1.cpp.o
[ 11%] Building CXX object dlib_build/CMakeFiles/dlib.dir/dir_nav/dir_nav_kernel_2.cpp.o
[ 11%] Building CXX object dlib_build/CMakeFiles/dlib.dir/dir_nav/dir_nav_extensions.cpp.o
[ 12%] Building CXX object dlib_build/CMakeFiles/dlib.dir/gui_widgets/fonts.cpp.o
[ 13%] Building CXX object dlib_build/CMakeFiles/dlib.dir/linker/linker_kernel_1.cpp.o
[ 13%] Building CXX object dlib_build/CMakeFiles/dlib.dir/logger/extra_logger_headers.cpp.o
[ 14%] Building CXX object dlib_build/CMakeFiles/dlib.dir/logger/logger_kernel_1.cpp.o
[ 14%] Building CXX object dlib_build/CMakeFiles/dlib.dir/logger/logger_config_file.cpp.o
[ 15%] Building CXX object dlib_build/CMakeFiles/dlib.dir/misc_api/misc_api_kernel_1.cpp.o
[ 16%] Building CXX object dlib_build/CMakeFiles/dlib.dir/misc_api/misc_api_kernel_2.cpp.o
[ 16%] Building CXX object dlib_build/CMakeFiles/dlib.dir/sockets/sockets_extensions.cpp.o
[ 17%] Building CXX object dlib_build/CMakeFiles/dlib.dir/sockets/sockets_kernel_2.cpp.o
[ 18%] Building CXX object dlib_build/CMakeFiles/dlib.dir/sockstreambuf/sockstreambuf.cpp.o
[ 18%] Building CXX object dlib_build/CMakeFiles/dlib.dir/sockstreambuf/sockstreambuf_unbuffered.cpp.o
[ 19%] Building CXX object dlib_build/CMakeFiles/dlib.dir/server/server_kernel.cpp.o
[ 20%] Building CXX object dlib_build/CMakeFiles/dlib.dir/server/server_iostream.cpp.o
[ 20%] Building CXX object dlib_build/CMakeFiles/dlib.dir/server/server_http.cpp.o
[ 21%] Building CXX object dlib_build/CMakeFiles/dlib.dir/threads/multithreaded_object_extension.cpp.o
[ 22%] Building CXX object dlib_build/CMakeFiles/dlib.dir/threads/threaded_object_extension.cpp.o
[ 22%] Building CXX object dlib_build/CMakeFiles/dlib.dir/threads/threads_kernel_1.cpp.o
[ 23%] Building CXX object dlib_build/CMakeFiles/dlib.dir/threads/threads_kernel_2.cpp.o
[ 24%] Building CXX object dlib_build/CMakeFiles/dlib.dir/threads/threads_kernel_shared.cpp.o
[ 24%] Building CXX object dlib_build/CMakeFiles/dlib.dir/threads/thread_pool_extension.cpp.o
[ 25%] Building CXX object dlib_build/CMakeFiles/dlib.dir/threads/async.cpp.o
[ 25%] Building CXX object dlib_build/CMakeFiles/dlib.dir/timer/timer.cpp.o
[ 26%] Building CXX object dlib_build/CMakeFiles/dlib.dir/stack_trace.cpp.o
[ 27%] Building CXX object dlib_build/CMakeFiles/dlib.dir/cuda/cpu_dlib.cpp.o
[ 27%] Building CXX object dlib_build/CMakeFiles/dlib.dir/cuda/tensor_tools.cpp.o
[ 28%] Building CXX object dlib_build/CMakeFiles/dlib.dir/data_io/image_dataset_metadata.cpp.o
[ 29%] Building CXX object dlib_build/CMakeFiles/dlib.dir/data_io/mnist.cpp.o
[ 29%] Building CXX object dlib_build/CMakeFiles/dlib.dir/data_io/cifar.cpp.o
[ 30%] Building CXX object dlib_build/CMakeFiles/dlib.dir/global_optimization/global_function_search.cpp.o
[ 31%] Building CXX object dlib_build/CMakeFiles/dlib.dir/filtering/kalman_filter.cpp.o
In file included from /media/mendel/dlib/dlib/filtering/../matrix.h:11,
                 from /media/mendel/dlib/dlib/filtering/kalman_filter.h:7,
                 from /media/mendel/dlib/dlib/filtering/kalman_filter.cpp:6:
/media/mendel/dlib/dlib/filtering/../matrix/matrix_la.h: In function ‘long int dlib::svd4(dlib::svd_u_mode, bool, const dlib::matrix_exp<EXP>&, dlib::matrix<typename EXP::type, uM, uN, MM1, L1>&, dlib::matrix<typename EXP::type, qN, qX, MM2, L1>&, dlib::matrix<typename EXP::type, vM, vN, MM3, L1>&) [with EXP = dlib::matrix_op<dlib::op_trans<dlib::matrix_op<dlib::op_trans<dlib::matrix<double, 1, 2, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout> > > > >; long int qN = 1; long int qX = 1; long int uM = 1; long int uN = 1; long int vM = 2; long int vN = 1; MM1 = dlib::memory_manager_stateless_kernel_1<char>; MM2 = dlib::memory_manager_stateless_kernel_1<char>; MM3 = dlib::memory_manager_stateless_kernel_1<char>; L1 = dlib::row_major_layout]’:
/media/mendel/dlib/dlib/filtering/../matrix/matrix_la.h:225:32: warning: iteration 1 invokes undefined behavior [-Waggressive-loop-optimizations]
             y = abs(q(i)) + abs(e(i));
                             ~~~^~~~~~
/media/mendel/dlib/dlib/filtering/../matrix/matrix_la.h:163:20: note: within this loop
         for (i=0; i<n; i++)
                   ~^~
[ 31%] Building CXX object dlib_build/CMakeFiles/dlib.dir/svm/auto.cpp.o
[ 32%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/arm/arm_init.c.o
[ 33%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/arm/filter_neon_intrinsics.c.o
[ 33%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/png.c.o
[ 34%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngerror.c.o
[ 34%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngget.c.o
[ 35%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngmem.c.o
[ 36%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngpread.c.o
[ 36%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngread.c.o
[ 37%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngrio.c.o
[ 38%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngrtran.c.o
[ 38%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngrutil.c.o
[ 39%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngset.c.o
[ 40%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngtrans.c.o
[ 40%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngwio.c.o
[ 41%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngwrite.c.o
[ 42%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngwtran.c.o
[ 42%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libpng/pngwutil.c.o
[ 43%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/adler32.c.o
[ 44%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/compress.c.o
[ 44%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/crc32.c.o
[ 45%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/deflate.c.o
[ 45%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/gzclose.c.o
[ 46%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/gzlib.c.o
[ 47%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/gzread.c.o
[ 47%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/gzwrite.c.o
[ 48%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/infback.c.o
[ 49%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/inffast.c.o
[ 49%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/inflate.c.o
[ 50%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/inftrees.c.o
[ 51%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/trees.c.o
[ 51%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/uncompr.c.o
[ 52%] Building C object dlib_build/CMakeFiles/dlib.dir/external/zlib/zutil.c.o
[ 53%] Building CXX object dlib_build/CMakeFiles/dlib.dir/image_loader/png_loader.cpp.o
[ 53%] Building CXX object dlib_build/CMakeFiles/dlib.dir/image_saver/save_png.cpp.o
[ 54%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jaricom.c.o
[ 54%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcapimin.c.o
[ 55%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcapistd.c.o
[ 56%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcarith.c.o
[ 56%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jccoefct.c.o
[ 57%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jccolor.c.o
[ 58%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcdctmgr.c.o
[ 58%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jchuff.c.o
[ 59%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcinit.c.o
[ 60%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcmainct.c.o
[ 60%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcmarker.c.o
[ 61%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcmaster.c.o
[ 62%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcomapi.c.o
[ 62%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcparam.c.o
[ 63%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcprepct.c.o
[ 64%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jcsample.c.o
[ 64%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdapimin.c.o
[ 65%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdapistd.c.o
[ 65%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdarith.c.o
[ 66%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdatadst.c.o
[ 67%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdatasrc.c.o
[ 67%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdcoefct.c.o
[ 68%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdcolor.c.o
[ 69%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jddctmgr.c.o
[ 69%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdhuff.c.o
[ 70%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdinput.c.o
[ 71%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdmainct.c.o
[ 71%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdmarker.c.o
[ 72%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdmaster.c.o
[ 73%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdmerge.c.o
[ 73%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdpostct.c.o
[ 74%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jdsample.c.o
[ 74%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jerror.c.o
[ 75%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jfdctflt.c.o
[ 76%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jfdctfst.c.o
[ 76%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jfdctint.c.o
[ 77%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jidctflt.c.o
[ 78%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jidctfst.c.o
[ 78%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jidctint.c.o
[ 79%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jmemmgr.c.o
[ 80%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jmemnobs.c.o
[ 80%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jquant1.c.o
[ 81%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jquant2.c.o
[ 82%] Building C object dlib_build/CMakeFiles/dlib.dir/external/libjpeg/jutils.c.o
[ 82%] Building CXX object dlib_build/CMakeFiles/dlib.dir/image_loader/jpeg_loader.cpp.o
[ 83%] Building CXX object dlib_build/CMakeFiles/dlib.dir/image_saver/save_jpeg.cpp.o
[ 84%] Linking CXX static library libdlib.a
[ 84%] Built target dlib
Scanning dependencies of target _dlib_pybind11
[ 84%] Building CXX object CMakeFiles/_dlib_pybind11.dir/src/dlib.cpp.o
In file included from /media/mendel/dlib/dlib/external/pybind11/include/pybind11/pytypes.h:12,
                 from /media/mendel/dlib/dlib/external/pybind11/include/pybind11/cast.h:13,
                 from /media/mendel/dlib/dlib/external/pybind11/include/pybind11/attr.h:13,
                 from /media/mendel/dlib/dlib/external/pybind11/include/pybind11/pybind11.h:43,
                 from /media/mendel/dlib/dlib/../dlib/python/pybind_utils.h:6,
                 from /media/mendel/dlib/dlib/../dlib/python.h:6,
                 from /media/mendel/dlib/tools/python/src/opaque_types.h:6,
                 from /media/mendel/dlib/tools/python/src/dlib.cpp:4:
/media/mendel/dlib/dlib/external/pybind11/include/pybind11/detail/common.h:112:10: fatal error: Python.h: No such file or directory
 #include <Python.h>
          ^~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/_dlib_pybind11.dir/build.make:63: CMakeFiles/_dlib_pybind11.dir/src/dlib.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/_dlib_pybind11.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
Traceback (most recent call last):
  File "setup.py", line 261, in <module>
    'Topic :: Software Development',
  File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3/dist-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/usr/lib/python3/dist-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3/dist-packages/setuptools/command/bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/usr/lib/python3/dist-packages/setuptools/command/bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3/dist-packages/setuptools/command/install_lib.py", line 24, in run
    self.build()
  File "/usr/lib/python3.7/distutils/command/install_lib.py", line 109, in build
    self.run_command('build_ext')
  File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 134, in run
    self.build_extension(ext)
  File "setup.py", line 174, in build_extension
    subprocess.check_call(cmake_build, cwd=build_folder)
  File "/usr/lib/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '-j1']' returned non-zero exit status 2.

```

**Comment by goruck** [comment]
@cappittall thanks for your patience on my reply. 

(1) regarding your first issue related to swap. Did you ```cd /media/mendel``` before creating and activating swap on the SD card? Below is information from my system.
```text
endel@tpu:/media/mendel$ ls -l
total 1048636
drwxr-xr-x  5 mendel mendel       4096 May 25  2020 calibrate_camera
drwxr-xr-x  7 mendel mendel       4096 Oct 31  2020 detection-server
drwxr-xr-x  4 mendel mendel       4096 May 11  2020 detect-server
drwxr-xr-x 10 mendel mendel       4096 Mar 15  2020 dlib
drwxr-xr-x  5 mendel mendel       4096 Apr 13  2020 google-coral
drwxr-xr-x  3 mendel mendel       4096 Mar 18  2020 grpc
drwx------  2 mendel mendel      16384 Mar 24  2019 lost+found
drwxr-xr-x 12 mendel mendel       4096 Mar 15  2020 opencv-3.4.5
drwxr-xr-x  6 mendel mendel       4096 Sep 22  2019 opencv_contrib-3.4.5
drwxr-xr-x  6 mendel mendel       4096 May 11  2020 sem-seg-server
-rw-------  1 root   root   1073741824 Mar 15  2020 swapfile
drwxr-xr-x  5 mendel mendel       4096 Jan 22  2020 tpu-servers
mendel@tpu:/media/mendel$ df
Filesystem                                                                   1K-blocks       Used  Available Use% Mounted on
/dev/root                                                                      7294440    3749676    3187272  55% /
devtmpfs                                                                        339300          0     339300   0% /dev
tmpfs                                                                           503684          0     503684   0% /dev/shm
tmpfs                                                                           503684      23668     480016   5% /run
tmpfs                                                                             5120          4       5116   1% /run/lock
tmpfs                                                                           503684          0     503684   0% /sys/fs/cgroup
tmpfs                                                                           503684        412     503272   1% /var/log
/dev/mmcblk0p1                                                                  126931      29677      90701  25% /boot
/dev/mmcblk1                                                                  61126476    2360104   55631604   5% /media/mendel
tmpfs                                                                           100736       4708      96028   5% /run/user/1000
lindo@192.168.1.4:/nvr                                                      3844639560 1602013152 2047259200  44% /mnt/nvr
lindo@192.168.1.4:/home/lindo/develop/smart-zoneminder/face-det-rec/dataset  477909784  175979080  277584492  39% /mnt/dataset
mendel@tpu:/media/mendel$ 
```
(2) regarding your second issue with dlib. I'm not exactly sure why the compiler fails. You are using dlib v19.22 and Python 3.7 but I used dlib v19.17 and Python 3.5. So its possible the README is incorrect for your combination. I am upgrading my system to your versions or later and will update this project's README accordingly.

**Comment by cappittall** [comment]
Hello @goruck ,

Thanks for the reply.  I  also appologies for the late answer. I didn' see the alert. However. 
1. Yes I did exactly the way you write.  And Unfortunately its the same. But I solved the problem with less amounth of swap size
2. Solved the second problem with installing one of neccessary files, you mentioned at opencv installation. But I don't know which one is exactly.  Maybe '# Install neon SIMD acceleration dependencies.'

Again thank you very much for kind assistance.

**Comment by cappittall** [comment]
Hello @goruck ,

Thanks for the reply. I also appologies for the late answer. I didn' see the alert. However.

Yes I did exactly the way you write. And Unfortunately its the same. But I solved the problem with less amounth of swap size
Solved the second problem with installing one of neccessary files, you mentioned at opencv installation. But I don't know which one is exactly. Maybe '# Install neon SIMD acceleration dependencies.'
Again thank you very much for kind assistance.
